okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/metrics/HostMetrics.java

```diff
@@ -16,26 +16,58 @@
 
 package com.palantir.remoting3.okhttp.metrics;
 
+import static com.google.common.base.Preconditions.checkState;
+
 import com.codahale.metrics.Meter;
 import com.codahale.metrics.MetricRegistry;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.tritium.tags.TaggedMetric;
+import java.util.Map;
+import java.util.Optional;
+import java.util.SortedMap;
 
 public final class HostMetrics {
 
+    public static final String SERVICE_NAME_TAG = "service-name";
+    public static final String HOSTNAME_TAG = "hostname";
+    public static final String FAMILY_TAG = "family";
+
     private final Meter informational;
     private final Meter successful;
     private final Meter redirection;
     private final Meter clientError;
     private final Meter serverError;
     private final Meter other;
 
-    public HostMetrics(MetricRegistry registry, String name, String hostname) {
-        // TODO(jellis): #581 add hostname as a tag
-        informational = registry.meter(MetricRegistry.name(name, "response", "family", "informational"));
-        successful    = registry.meter(MetricRegistry.name(name, "response", "family", "successful"));
-        redirection   = registry.meter(MetricRegistry.name(name, "response", "family", "redirection"));
-        clientError   = registry.meter(MetricRegistry.name(name, "response", "family", "client-error"));
-        serverError   = registry.meter(MetricRegistry.name(name, "response", "family", "server-error"));
-        other         = registry.meter(MetricRegistry.name(name, "response", "family", "other"));
+    public HostMetrics(MetricRegistry registry, String serviceName, String hostname) {
+        informational = registry.meter(name(serviceName, hostname, "informational"));
+        successful    = registry.meter(name(serviceName, hostname, "successful"));
+        redirection   = registry.meter(name(serviceName, hostname, "redirection"));
+        clientError   = registry.meter(name(serviceName, hostname, "client-error"));
+        serverError   = registry.meter(name(serviceName, hostname, "server-error"));
+        other         = registry.meter(name(serviceName, hostname, "other"));
+    }
+
+    private static String name(String serviceName, String hostname, String family) {
+        Map<String, String> tags = ImmutableMap.<String, String>builder()
+                .put(SERVICE_NAME_TAG, serviceName)
+                .put(HOSTNAME_TAG, hostname)
+                .put(FAMILY_TAG, family)
+                .build();
+        return TaggedMetric.toCanonicalName("client.response", tags);
+    }
+
+    public static Optional<Meter> getMeter(
```

Comment:
this is used in tests only, right? would prefer to move it into the test scope

Reply:
moved into `HostMetricsTest`